app/controllers/manage_evaluators_controller.rb

```diff
@@ -0,0 +1,179 @@
+# frozen_string_literal: true
+
+class ManageEvaluatorsController < ApplicationController
+  include ManageEvaluatorsHelper
+
+  before_action :set_challenge
+
+  VALID_EVALUATOR_ROLES = %w[evaluator solver challenge_manager].freeze
+
+  def index
+    @phases = @challenge.phases.order(:start_date)
+
+    if @phases.empty?
+      handle_empty_phases
+    else
+      handle_existing_phases
+    end
+  end
+
+  def create
+    @phase = @challenge.phases.find(evaluator_invitation_params[:phase_id])
+    result = process_evaluator_invitation(evaluator_invitation_params[:email])
+
+    if result[:success]
+      handle_successful_creation(result)
+    else
+      handle_failed_creation
+    end
+  end
+
+  def destroy
+    @phase = @challenge.phases.find(params[:phase_id])
+    result = process_evaluator_removal(params[:evaluator_type], params[:evaluator_id])
```

Comment:
Rails route adjustment
```suggestion
    result = process_evaluator_removal(params[:evaluator_type], params[:id])
```